src/south/ocnos/goldstone/south/ocnos/vlan.py

```diff
@@ -0,0 +1,384 @@
+            xpath_customer_vlan, value=value
+        )
+
+        insert_xml(xml_customer_vlan, xml_bridge_vlan, TAG_IPI_VLAN)
+
+        config = new_ele("config")
+        config.append(xml_bridge_vlan)
+        logger.debug(
+            f"Sending xml for vlan id configuration: {etree.tostring(config,pretty_print=True).decode()}"
+        )
+        return self.server.ocnos_conn.conn.edit_config(config)
+
+    def unset_vlan_name(self, user):
+        cache = self.setup_cache(user)
+
+        # Check vlan-id is present on candidate-config before removing the vlans's name.
+        # It's necessary because the sysrepo try to remove the vlan-name also, when the
+        # vlan-id is being removed. If vlan-id is not present it is not necessary
+        # to send the name removal to the OcNOS.
+        xpath_list = list(libyang.xpath_split(self.change.xpath))
+        assert xpath_list[1][1] == "vlan"
+        vlan_id = xpath_list[1][2][0][1]
+        xpath_gs_vlan = GS_VLAN.format(vlan_id)
+        if not libyang.xpath_get(cache, xpath_gs_vlan, None):
+            return
+
+        xpath_bridge_vlan = IPI_BRIDGE_VLAN.format(
+            INSTANCE_NAME_FOR_IEEE_VLAN_BRIDGE, INSTANCE_TYPE, self.vlan_id
+        )
+        xpath_customer_vlan = IPI_CUSTOMER_VLAN_CONFIG_NAME
+        xml_bridge_vlan = self.server.ocnos_conn.xpath2xml(xpath_bridge_vlan)
+        xml_customer_vlan = self.server.ocnos_conn.xpath2xml(
+            xpath_customer_vlan, delete_oper=True
+        )
+
+        insert_xml(xml_customer_vlan, xml_bridge_vlan, TAG_IPI_VLAN)
+
+        config = new_ele("config")
+        config.append(xml_bridge_vlan)
+        logger.debug(
+            f"Sending xml for vlan id configuration: {etree.tostring(config,pretty_print=True).decode()}"
+        )
+        return self.server.ocnos_conn.conn.edit_config(config)
+
+
+class VlanServer(ServerBase):
+    def __init__(self, sr_conn, ocnos_conn):
+        super().__init__(sr_conn, "goldstone-vlan")
+        self.sr_conn = sr_conn
+        self.ocnos_conn = ocnos_conn
+
+        # Create default network-instance
+        self._create_default_network_instance(
+            INSTANCE_NAME_FOR_IEEE_VLAN_BRIDGE, INSTANCE_TYPE
+        )
+
+        # TODO: other handlers should be implemented in future.
+        self.handlers = {
+            "vlans": {
+                "vlan": {
+                    "vlan-id": NoOp,
+                    "config": {"vlan-id": VlanIdHandler, "name": VlanNameHandler},
+                }
+            }
+        }
+
+    async def start(self):
+        await self.reconcile()
+        tasks = await super().start()
+
+        return tasks
+
+    def stop(self):
+        # Remove default network-instance
+        self._delete_default_network_instance(
+            INSTANCE_NAME_FOR_IEEE_VLAN_BRIDGE, INSTANCE_TYPE
+        )
+
+        super().stop()
+
+    def pre(self, user):
+        # TODO
+        pass
+
+    async def post(self, user):
+        # TODO
```

Comment:
what is TODO here?

Reply:
Sorry, just forgot to delete the TODO line.
Deleted it.

Reply:
please check other similar places.

https://github.com/oopt-goldstone/goldstone-mgmt/pull/71/files#diff-64dd4da385d6884571543f3fee097fba863d284e8a6dee1ae5f2b7ac2f03a8f6R87-R92

Reply:
Thank you for pointing it out. I checked similar places and deleted unnecessary lines.